### ISSUE TITLE
inlined function not getting deleted

### ISSUE THREAD

**Issue by flocsy** [markw65/monkeyc-optimizer#14]
```
import Toybox.Lang;
import Toybox.WatchUi;

class OptiBug {
    var labelUnitString as String;
    var label as String;

    function initialize(measureId as Number) {
        self.labelUnitString = "foo";
        self.label = getLabel();
    }

    (:inline)
    hidden function getLabel() as String {
        var labelSuffix = self.labelUnitString;
        return (WatchUi.loadResource(Rez.Strings.AppName) as String) + labelSuffix;
    }
}

```

Even though getLabel() gets correctly inlined, the original method is kept (and the only place it is used is in the initialize)

**Comment by markw65** [comment]
This example works for me, even with `Iterate Optimizer` disabled.

Is it possible that you have another method called getLabel (in another class or module) that doesn't get eliminated? The optimizer is quite conservative when it comes to eliminating class methods - it won't eliminate it if it sees a possible reference from elsewhere (and yes, I realize that "possible references" from outside the class can't actually reference it, because its hidden).

**Comment by flocsy** [comment]
Yes I have getLabel() in my barrel. So can you fix this? Or is there a workaround?

**Comment by markw65** [comment]
> So can you fix this

Yes, and I hope to do so. But doing it correctly is a fairly large project, and I may not get around to it for a while.

> well the obvious one worked
Yes

> However a warning similar to the blue...
That should be easy. I can just issue a diagnostic for any inline function that gets output. It doesn't take any analysis.
